doc/apiary/v2/fiware-ngsiv2-reference.apib

```diff
@@ -566,78 +566,104 @@ The query returns all the entities that match all the matching conditions (AND l
 
 There are two kinds of statements: *unary statements* and *binary statements*.
 
-Binary statements are composed by an attribute name (e.g. "temperature"), an operator and a value
+Binary statements are composed by an attribute path (e.g. `temperature`), an operator and a value
 (whose format depends on the operator), e.g.:
 
 ```
 temperature==50
 temperature<=20
 ```
 
+The syntax of an attribute path consists of a list of tokens separated by the `.` character. This list of tokens
+addresses a JSON property name, in accordance with the following rules:
+
+* The first token is the name of an NGSI attribute (*target NGSI attribute*) of an entity.
+* If filtering by attribute value (i.e. the expression is used in a `q` query), the rest of tokens (if present)
+  represent the path to a sub-property of the *target NGSI attribute value* (which should be a JSON object).
+  Such sub-property is defined as the *target property*.
+* If filtering by metadata (i.e. the expression is used in a `q` query), the second token represents a metadata
+  name associated to the target NGSI attribute, *target metadata*, and the rest of tokens
+  (if present) represent the path to a sub-property of the *target metadata value* (which should be a
+  JSON object). Such sub-property is defined as the *target property*.
+
+The *target property value* is defined as the value of the JSON property addressed by the list of tokens described
+above i.e. the value of the *target property*.
+
+In case only one token is provided (two in the case of filtering by metadata), then the *target property* will
+be the target NGSI attribute itself (or the target metadata in the case of filtering by metadata) and the
+*target property value* will be the target NGSI attribute value (or the target metadata value in the case
+of filtering by metadata). It is assumed that the value of the target NGSI attribute (or the target metadata
```

Comment:
However, in the other case (when we have more than one/two token) we are assuming the the value of the attribute/metadata is a JSON object. Literally: "(which should be a JSON object)". Thus, it makes sense to do the opposite assumption here (maybe using the same style, i.e. "it shouldn't be a JSON object" to make the parallelism more explicit).

Having said that, I don't like this paragraph too much. However, it is the best way I have find to describe the special case of "no navigation". Alternatives are welcomed!